video/out/hwdec/hwdec_vaapi.c

```diff
@@ -275,6 +281,24 @@ static int mapper_map(struct ra_hwdec_mapper *mapper)
     CHECK_VA_STATUS(mapper, "vaSyncSurface()");
     p->surface_acquired = true;
 
+    if (wldmabuf) {
+        close_file_descriptors(desc);
+        /* composed has single layer */
+        int layer_no = 0;
+        uint32_t drm_format = desc.layers[layer_no].drm_format;
+
+        if (!ra_compatible_format(mapper->ra, drm_format, desc.objects[0].drm_format_modifier)) {
+            MP_DBG(mapper->owner, "Mapped surface with format %s; drm format '%s(%016lx)', is "
+                                  "not supported by compositor.\n",
+                                   mp_imgfmt_to_name(mapper->src->params.hw_subfmt),
+                                   mp_tag_str(drm_format),
+                                   desc.objects[0].drm_format_modifier);
+            goto err;
```

Comment:
I'm not sure how exactly this would be be done, but I feel like hitting this error condition should propagate to dmabuf-wayland and cause it to error out. Or well if you know that you can't map any surfaces at all, then there's no point in continuing.

Reply:
This error condition is normal - it just means that this particular surface is not supported.

Reply:
Yes, if no surfaces can be mapped, and the ra is wldmabuf, then we should error out.

Reply:
I've added this sanity check.